plugins/transfer/plugin.go

```diff
@@ -130,6 +130,11 @@ func init() {
 					return nil, fmt.Errorf("no matching diff plugins: %w", errdefs.ErrNotFound)
 				}
 
+				// If CheckPlatformSupported is false, we will match all platforms
+				if !config.CheckPlatformSupported {
```

Comment:
@dmcgowan   I added this config to transfer plugin to optionally not enforcing strict platform matching in unpackConfiguration `platform, snapshotter` due to [cri-tools pull test](https://github.com/kubernetes-sigs/cri-tools/blob/master/test/e2e/pull_test.go#L57) that pulls an image digest which is arm64 arch based.   
This test failed on the CI with the default platform(amd64) and snapshotter combination.    
Does it make sense to have such config here?    or should we update containerd config file to pass the test in this case? 

My understanding is with the defaultUnpackConfiguration, i.e. `amd64,overlayfs`,  transfer service doesn't allow unpacking different arch(arm64) of image other than the running platform unless we explicitly add that UnpackConfiguration `arm64, overlayfs` to the supported platform. This is different with the client.pull approach.

Reply:
We really need a better platform matcher for that case. For unpack configuration it should really be able to just match OS, the architecture does not matter too much for unpacker. Where is it enforcing this check today, if the image itself is pulled by digest.